tests/integration/s2n_test_constants.py

```diff
@@ -243,7 +243,7 @@
         client_sni=None,
         client_ciphers="ECDHE-RSA-AES128-SHA",
         expected_cert=SNI_CERTS["many_animals"],
-        expect_matching_hostname=True),
+        expect_matching_hostname=False),
```

Comment:
fyi this was passing previously because we had `client_sni=None` and s_client doesn't complain about domain mismatch if you don't send an SNI.